### ISSUE TITLE
kustomize edit add configmap illegally repeats the key

### ISSUE THREAD

**Issue by MaurGi** [kubernetes-sigs/kustomize#5217]
### What happened?

Calling `kustomize edit add configmap deployer-info --from-literal=test=value` twice errors out:
Error: `configmap deployer-info illegally repeats the key 'test'`

### What did you expect to happen?

The script should be idempotent and just replace the value rather than erroring out.

### How can we reproduce it (as minimally and precisely as possible)?

- create an empty kustomization.yaml file.
- run `kustomize edit add configmap deployer-info --from-literal=test=value`
- run `kustomize edit add configmap deployer-info --from-literal=test=value`

### Expected output

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
configMapGenerator:
- literals:
  - test=value
  name: deployer-info
```

### Actual output

Error:
```
Error: configmap deployer-info illegally repeats the key `test`
Usage:
  kustomize edit add configmap NAME [--behavior={create|merge|replace}] [--from-file=[key=]source] [--from-literal=key1=value1] [flags]

Examples:

        # Adds a configmap to the kustomization file (with a specified key)
        kustomize edit add configmap my-configmap --from-file=my-key=file/path --from-literal=my-literal=12345

        # Adds a configmap to the kustomization file (key is the filename)
        kustomize edit add configmap my-configmap --from-file=file/path

        # Adds a configmap from env-file
        kustomize edit add configmap my-configmap --from-env-file=env/path.env

        # Adds a configmap from env-file with behavior merge
        kustomize edit add configmap my-configmap --behavior=merge --from-env-file=env/path.env


Flags:
      --behavior string            Specify the behavior for config map generation, i.e whether to create a new configmap (the default),  to merge with a previously defined one, or to replace an existing one. Merge and replace should be used only  when overriding an existing configmap defined in a base
      --disableNameSuffixHash      Disable the name suffix for the configmap
      --from-env-file string       Specify the path to a file to read lines of key=val pairs to create a configmap (i.e. a Docker .env file).
      --from-file strings          Key file can be specified using its file path, in which case file basename will be used as configmap key, or optionally with a key and file path, in which case the given key will be used.  Specifying a directory will iterate each named file in the directory whose basename is a valid configmap key.
      --from-literal stringArray   Specify a key and literal value to insert in configmap (i.e. mykey=somevalue)
  -h, --help                       help for configmap

Global Flags:
      --stack-trace   print a stack-trace on error
```

### Kustomize version

v5.0.3

### Operating system

Linux

**Comment by natasha41575** [comment]
/assign @stormqueen1990

I _think_ throwing an error is the right behavior here. Theoretically, we should have a `kustomize edit set configmap` command that has the behavior that you're describing. I'm assigning this to @stormqueen1990 to take a look. Could you please investigate:

- whether or not `kustomize edit set configmap` exists
- its behavior, and whether or not it does what is requested in this issue
- if it doesn't exist, what other `kustomize edit set` commands do we have, and does it make sense to add one

**Comment by stormqueen1990** [comment]
> Could you please investigate:
> 
> * whether or not `kustomize edit set configmap` exists

`kustomize edit set configmap` does not currently exist.

> * if it doesn't exist, what other `kustomize edit set` commands do we have, and does it make sense to add one

`kustomize edit set` supports the following subcommands:
* `annotation`
* `buildmetadata`
* `label`
* `nameprefix`
* `namespace`
* `namesuffix`
* `replicas`

Considering that it's possible to remove and edit a label using `kustomize`, I think it could make sense from a code parity standpoint to add such commands for ConfigMaps and Secrets as well.

**Comment by stormqueen1990** [comment]
I took another look at this and noticed there's a flag `--behavior`, for which the options are `create` (default), `merge`, and `replace`. However, issuing `kustomize edit add configmap <configmap-name> --behavior=merge --from-literal=key=value` still results in an error message saying that the key is being illegally repeated.

**Comment by stormqueen1990** [comment]
It seems the `behavior` flag is for the actual ConfigMap creation and not for the kustomization, so `kustomize edit set`/`kustomize edit remove` would still be the route to allow editing and removing ConfigMaps from a kustomization file.

**Comment by natasha41575** [comment]
Sg. I also found https://github.com/kubernetes-sigs/kustomize/issues/5233, which as one of its bulletpoints requests to add `kustomize edit set configmap`. I think we should leave the behavior of `kustomize edit add configmap` as-is, and we can implement `kustomize edit set configmap` for this use case. Thank you @stormqueen1990 for your invetigation and writeup. Closing this, we can use https://github.com/kubernetes-sigs/kustomize/issues/5233 for further tracking/`edit` enhacements.
